activerecord/test/cases/serialized_attribute_test.rb

```diff
@@ -502,12 +502,12 @@ def test_decorated_type_with_decorator_block
     klass = Class.new(ActiveRecord::Base) do
       self.table_name = Topic.table_name
       store :content, coder: ActiveRecord::Coders::JSON
-      attribute(:content) { |subtype| EncryptedType.new(subtype: subtype) }
+      attribute(:content, EncryptedType.new(subtype: Topic.attribute_types.fetch("content")))
```

Comment:
Is this required change? Actually this test case tests `attribute` with adecorator block in accordance with its test name.

https://github.com/rails/rails/blob/d5c88d68c31561b3597931f383dcc0a257f7d1e0/activerecord/test/cases/serialized_attribute_test.rb#L501

Reply:
> Is this required change? 

The `attribute` method doesn't take a block and never yields. I'll go check if it did when this test was introduced, but I don't think it ever did. So this test was essentially a noop.

Reply:
This internal feature and test was introduced in #41139 by me.

Reply:
Right, but as of https://github.com/rails/rails/pull/44666 it no longer exist. If it was never public API, then I suppose we don't need to warn or raise.

I updated the test to use the new API.

Reply:
Yeah, I didn't know that had been replaced to `decorate_attributes`, thanks for the quick investigation!

Reply:
> I updated the test to use the new API.

Somehow forgot to push... I'll open another PR.